src/App.js

```diff
@@ -1,13 +1,21 @@
-import React from 'react';
-import { Route, Switch } from 'react-router-dom';
+import React, { useState } from 'react';
+import { Route, Switch, Redirect } from 'react-router-dom';
 import './App.css';
-import Test from './components/Test';
 import Navigation from './components/Navigation';
 import AddItem from './components/AddItem';
 import List from './components/List';
 import Welcome from './components/Welcome';
+import getToken from './lib/tokens';
 
 function App() {
+  const [token, setToken] = useState(localStorage.getItem('userToken') || '');
+
+  const handleClick = () => {
+    const token = getToken();
+    localStorage.setItem('userToken', token);
+    setToken(token);
+  };
+
```

Comment:
@veeteeran  and I discussed creating a `useLocalStorage` custom hook here. Do others agree this would be a good update to make the code more reusable or would this be adding unnecessary complexity? If it's a good idea, should the custom hook set and get the token, or just set it.

Reply:
I'm not sure if we need a custom hook. If we can ensure that the user token exists before the user goes to the Add Item page then we can pull the user token directly from localStorage.

Reply:
Okay, awesome!

Reply:
Yep, another way would be to pass the token down as a prop to the components that need it.

Reply:
I always find it super tough to decide when to add an "abstraction", which is what a custom hook would be here. I think that most devs (including me) jump to making abstractions because, well, it's fun! It's also super tempting to want to pre-optimize our code.

In response to the traditional idea of "DRY Programming" DRY (Don't Repeat Yourself), there is a new trend among the cool kids of programming these days called "AHA Programming" (Avoid Hasty Abstraction). Here's a really great blog post about it from one of the React superstars - https://kentcdodds.com/blog/aha-programming 

Long story short, lately I've been trying to wait until I've repeated some code a bunch of times before I create an abstraction around it. By waiting, it usually gives you a much better idea of what the really problem that the abstraction is solving and how best to write it.

Reply:
Thanks @MikeBLambert that was a great take on abstraction. I haven't heard about AHA before and I want to be a cool kid! 😆 I feel like everyone tcl-17 is planning ahead. If we were to build out a task list with columns `urgent`, `do soon`, `great if we have time` this custom hook could be put in the last column for now. Didn't @laurenmbeatty have a slide like this during our first sync? 🤔

Reply:
It's definitely a challenge to find the right balance between DRY and AHA. I like the idea of waiting until you have a better idea of how best to solve the problem!